src/controller.cpp

```diff
@@ -931,6 +934,16 @@ bool Controller::activate(const ServerData& serverData,
       return true;
     }
 
+    // Ensure that the device is connected to the Internet.
+    if (MozillaVPN::instance()->networkWatcher()->getReachability() ==
+        QNetworkInformation::Reachability::Disconnected) {
+      logger.debug() << "Internet probe failed during controller activation. "
+                        "Device has no network connectivity.";
+      m_deviceNetworkConnectivity = false;
```

Comment:
Looks like m_deviceNetworkConnectivity is never set to true, if the network is restored.